ckan/logic/__init__.py

```diff
@@ -177,6 +179,15 @@ def parse_params(
         if len(value) == 1:
             value = value[0]
         parsed[key] = value
+    # checks and delete, if the csrf_token is in "parsed".
+    # we don't want the csrf_token to be a part of a data_dict
+    # as it will expose the token to the resource-metadata.
+    # this way we are deleting the token from every data_dict that fills
+    # from request.form instead of deleting it separately in every view.
+    # WTF_CSRF_FIELD_NAME is added by flask_wtf
+    csrf_token = config.get_value("WTF_CSRF_FIELD_NAME")
+    if csrf_token in parsed:
+        parsed.pop(csrf_token)
```

Comment:
I usually don't like big comments since it's an indicator that the logic is too complex to understand. In this cases is better to extract the logic to a small method, add a descriptive name to it and use the docstring to document it. (instead of multiple comment lines.)

Can we refactor this into a cleaner function?

Reply:
sure, thanks for your note